analyzer/src/main/kotlin/managers/Npm.kt

```diff
@@ -240,7 +244,35 @@ open class Npm(
                     .url("https://registry.npmjs.org/$encodedName")
                     .build()
 
-                OkHttpClientHelper.execute(HTTP_CACHE_PATH, pkgRequest).use { response ->
+                OkHttpClientHelper.execute(HTTP_CACHE_PATH, pkgRequest) {
+                    val npmrcFile = getUserHomeDirectory().resolve(".npmrc")
+                    if (npmrcFile.isFile) {
+                        val httpRegex = Regex("^https?://.+$")
+
+                        npmrcFile.forEachLine { line ->
+                            val (key, value) = line.split('=', limit = 2).map { it.trim() }
+
+                            val proxyUrl = value.takeIf { it.matches(httpRegex) } ?: when (key) {
+                                "proxy" -> "http://$value"
+                                "https-proxy" -> "https://$value"
+                                else -> ""
+                            }
+
+                            if (proxyUrl.isNotEmpty()) {
+                                val url = URI(proxyUrl)
+                                proxy(Proxy(Proxy.Type.HTTP, InetSocketAddress(url.host, url.port)))
+                                proxyAuthenticator { _, response ->
+                                    val user = url.userInfo.substringBefore(':')
+                                    val password = url.userInfo.substringAfter(':')
+                                    val credential = Credentials.basic(user, password)
+                                    response.request().newBuilder()
+                                        .header("Proxy-Authorization", credential)
+                                        .build()
+                                }
```

Comment:
Shouldn't the foreach loop be aborted here?

Reply:
Yeah, I was a bit undecided about that. The "problem" is that OkHttp does not make a difference between *http* and *https* proxies like NPM does. To me it felt wrong to arbitrarily only choose the first of either, so I tried if the code works if I simply apply both (I have both `proxy` and `https-proxy` set in my `.npmrc`), and it does.